### PR TITLE
Improve `toolImage` to be able to set without image tag

### LINKED ISSUE #4047: Improve `toolImage` to be able to set without image tag
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, the usage of the `toolImage` is described [here](https://docs.pingcap.com/tidb-in-kubernetes/dev/backup-restore-overview#general-fields), however, one case is not covered, where users want to use the BR with the same version of TiKV but have to use their own image registry instead of `pingcap`.
Refer to: https://asktug.com/t/topic/94029/4
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Closes #4047 
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [x] Manual test 

The TiKV's version is v5.0.1
* Deploy `Backup` with `spec.toolImage: `, then check if  image of Pod is 'pingcap/br:v5.0.1'.
* Deploy `Backup` with `spec.toolImage: pingcap/br:v5.0.2`, then check if  image of Pod is 'pingcap/br:v5.0.2'.
* Deploy `Backup` with `spec.toolImage: myimage`, then check if  image of Pod is 'myimage: v5.0.1'.

Jobs of all `Backup` is completed.
* Deploy `Restore` with `spec.toolImage: `, then check if  image of Pod is 'pingcap/br:v5.0.1'.
* Deploy `Restore` with `spec.toolImage: pingcap/br:v5.0.2`, then check if  image of Pod is 'pingcap/br:v5.0.2'.
* Deploy `Restore` with `spec.toolImage: myimage`, then check if  image of Pod is 'myimage: v5.0.1'.

Jobs of all `Restore` is completed.
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Use tool image of the `TiKV`'s version if tool image of `BR` does not contains tag
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -1680,6 +1680,7 @@ type RestoreSpec struct {
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 	// ToolImage specifies the tool image used in `Restore`, which supports BR and TiDB Lightning images.
 	// For examples `spec.toolImage: pingcap/br:v4.0.8` or `spec.toolImage: pingcap/tidb-lightning:v4.0.8`
+	// For BR image, if it does not contain tag, Pod will use image 'ToolImage:${TiKV_Version}'.
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:1683]: `BackupSpec` also needs to update.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**